lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp

```diff
@@ -76,31 +77,38 @@ static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
           amf_app_desc_p, AMF_APP_UL_DATA_IND(received_message_p).nas_msg,
           AMF_APP_UL_DATA_IND(received_message_p).ue_id,
           AMF_APP_UL_DATA_IND(received_message_p).tai);
+      is_task_state_same = true;
+      force_ue_write     = true;
       break;
     case AMF_APP_INITIAL_CONTEXT_SETUP_RSP:
       amf_app_handle_initial_context_setup_rsp(
           amf_app_desc_p,
           &AMF_APP_INITIAL_CONTEXT_SETUP_RSP(received_message_p));
+      is_task_state_same = true;
       break;
     /* Handle PDU session Response from UE */
     case N11_CREATE_PDU_SESSION_RESPONSE:
       amf_app_handle_pdu_session_response(
           &N11_CREATE_PDU_SESSION_RESPONSE(received_message_p));
+      is_task_state_same = true;
       break;
     case AMF_APP_SUBS_AUTH_INFO_RESP:
       amf_nas_proc_authentication_info_answer(
           &AMF_APP_AUTH_RESPONSE_DATA(received_message_p));
+      is_task_state_same = true;
       break;
 
     case AMF_APP_DECRYPT_IMSI_INFO_RESP:
       amf_decrypt_imsi_info_answer(
           &AMF_APP_DECRYPT_IMSI_RESPONSE_DATA(received_message_p));
+      is_task_state_same = true;
```

Comment:
This should also `force_ue_write = true` scenario since UE state is changing before registration

Reply:
Addressed review comment